### PR TITLE
SC47945: Add domain tags to snowflake resources and update looker dashboard

### PR DESCRIPTION
**Pull request by mayconbordin**
### Value of doing this
So that we can correctly report costs associated with the DWH to the domains using them.

### What needs to be done

- Update the `domain` module to;
  - Create a generic `owner` tag including as values all super domains, squads and some generic values (eg. `fh`).
  - Create a `superdomain` tag including as values all super domains and some generic values (eg. `fh`).
  - Create a `squad` tag including as values all squads and some generic values (eg. `fh`).
  - Assign the tags to warehouses and databases.
  - Keep an eye on warehouses created outside squads or domains, as these should ideally be assigned to a domain or squad and at least to a generic owner.

- Update the Snowflake usage block in Looker to use the tags:
  - Create new views for the `TAGS` and `TAG_REFERENCES` views of the Snowflake usage DB.
  - Join these views to the relevant explores.
  - Update the costs dashboard to use the tags for filtering.

### Acceptance criteria
Using the Existing Looker snowflake dashboard or a new one, I can grasp the Snowflake costs  associated to each FH domain. This includes the newly-created resources for FE/CX, OTA and CA.

### CODE REVIEW

File: snowflake_usage_block.model.lkml [hunk 1]
```diff
@@ -36,6 +36,12 @@ explore: query_history {
     sql_on: ${query_history.warehouse_name} = ${_squad_warehouse_mapping.warehouse_name} ;;
     relationship: many_to_one
   }
+  join: tag_references {
```
Comment (inigofh) [inline snowflake_usage_block.model.lkml:39]: I'd suggest adding the join type:

```suggestion
  join: tag_references {
    type: left_outer
```

File: snowflake_usage_block.model.lkml [hunk 2]
```diff
@@ -36,6 +36,12 @@ explore: query_history {
...
+    sql_on:
+      ${query_history.database_name} = ${tag_references.object_database}
+      AND ${query_history.schema_name} = ${tag_references.object_schema} ;;
```
Comment (inigofh) [inline snowflake_usage_block.model.lkml:42]: Do we need something here like what you have on other joins?

```suggestion
    sql_on:
      ${query_history.database_name} = ${tag_references.object_database}
      AND ${query_history.schema_name} = ${tag_references.object_schema}
      AND ${tag_references.domain} = 'SCHEMA' ;;
```

File: snowflake_usage_block.model.lkml [hunk 3]
```diff
@@ -36,6 +36,12 @@ explore: query_history {
...
+    relationship: many_to_one
```
Comment (inigofh) [inline snowflake_usage_block.model.lkml:43]: Probably `many_to_many`, right?

```suggestion
    relationship: many_to_many
```

**Review by inigofh — CHANGES_REQUESTED** [review]

**Comment by inigofh** [comment]
Hey @mayconbordin, my suggestions to add the join type and to use many to many applied to all joins you created. My bad because I didn't specify it.

**Comment by mayconbordin** [comment]
> Hey @mayconbordin, my suggestions to add the join type and to use many to many applied to all joins you created. My bad because I didn't specify it.

It only applied to one join:

```
join: tag_references {
    type: left_outer
    sql_on:
      ${query_history.database_name} = ${tag_references.object_database}
      AND ${query_history.schema_name} = ${tag_references.object_schema}
      AND ${tag_references.domain} = 'SCHEMA' ;;
    relationship: many_to_many
  }
```

All the other joins are many_to_one.

**Review by inigofh — APPROVED** [review]

**Pull request merged**